### ISSUE TITLE
problem in sum_energy function in prediction_geodesic.py

### ISSUE THREAD

**Issue by VatsalBatra** [SUTDBrainLab/MGP-VAE#6]
hi @sarthak268 @shagunuppal 
in the function 
```
def sum_energy(z_collection):
	
	delta_e = torch.FloatTensor(1, NUM_SAMPLE_GEO_OUTPUT, NDIM).zero_().cuda()
	for i in range(1, N):
		delta_e += find_energy(z_collection[i-1].view(NUM_SAMPLE_GEO_OUTPUT, -1) ,z_collection[i].view(NUM_SAMPLE_GEO_OUTPUT, -1) ,z_collection[i+1].view(NUM_SAMPLE_GEO_OUTPUT, -1))

	# energy_arr: a float tensor of size = (num_frames) where each index corresponds to energy of each point
	energy_arr = compute_norm(Variable(delta_e))
	energy_sum = (torch.sum(energy_arr)).item()

	return energy_sum
```

This function is essentially computing sum of gradient of energies of all interpolated vectors and adding them to `delta_e` and then taking L2 norm.
But in ALGORITHM 1 of this [paper](https://arxiv.org/pdf/1711.08014.pdf) (I think this is what is implemented in the paper)adds the squared L2 norm of each vector's energy gradient.
![image](https://user-images.githubusercontent.com/39874883/103648256-950e0980-4f82-11eb-8af5-ee3aa4f78fc5.png)

Please let me know if there is any correction.

**Comment by sarthak268** [comment]
Thanks, corrected
